apps/omg_child_chain/lib/omg_child_chain/fees/file_adapter.ex

```diff
@@ -35,7 +35,7 @@ defmodule OMG.ChildChain.Fees.FileAdapter do
 
     with {:changed, file_updated_at} <- check_file_changes(path, recorded_file_updated_at),
          {:ok, content} <- File.read(path),
-         {:ok, fee_specs} <- FeeParser.parse(content) do
+         {:ok, fee_specs} <- JSONFeeParser.parse(content) do
```

Comment:
`JSONFeeParser.parse(content)` is returning a list of tuple errors 
`{:error, list({:error, parsing_error(), any(), non_neg_integer() | nil})}`
your else handling is matching this case into `error` var and further into
https://github.com/omisego/elixir-omg/pull/1332/files#diff-a3c546fa6ad28e56859c0ca99ddd9729R122

so `{:error, {:error, list({:error, parsing_error(), any(), non_neg_integer() | nil})}, state}`. just validating if you're OK with this - if you are, dialyzer specs seem to be wrong